src/v/datalake/serde_parquet_writer.cc

```diff
@@ -19,7 +19,6 @@ ss::future<writer_error> serde_parquet_writer::add_data_struct(
     auto group = std::get<serde::parquet::group_value>(
       std::move(conversion_result.value()));
     try {
-        co_await _mem_tracker.maybe_reserve_memory(approx_size, as);
         auto stats = co_await _writer.write_row(std::move(group));
         _buffered_bytes = stats.buffered_size;
         _flushed_bytes = stats.flushed_size;
```

Comment:
It doesn't seem like error handling has changed, so it's not related to this PR, but can you remind me: at this point if the system is out of memory, it's up to the scheduler to stop this translation, right? Does that mean that a single translation could reserve all the memory, fail to reserve next, and then be aborted by the scheduler?

Reply:
yes thats possible today but I was planning to fix the OOM handling a bit before the release. The idea is finish the translation as usual if at least flush limit bytes have accumulated / lag limit is hit at oom abort time so it can make forward progress. Generally this OOM scenario is lacking test coverage, I plan add fix that a bit too. (also hoping stress tests would add some coverage).